nominal/core/data_review.py

```diff
@@ -0,0 +1,134 @@
+
+
+@dataclass(frozen=True)
+class CheckViolation:
+    rid: str
+    check_rid: str
+    name: str
+    start: IntegralNanosecondsUTC
+    end: IntegralNanosecondsUTC | None
+    priority: checklist.Priority | None
+
+    @classmethod
+    def _from_conjure(cls, check_alert: scout_datareview_api.CheckAlert) -> CheckViolation:
+        return cls(
+            rid=check_alert.rid,
+            check_rid=check_alert.check_rid,
+            name=check_alert.name,
+            start=_SecondsNanos.from_api(check_alert.start).to_nanoseconds(),
+            end=_SecondsNanos.from_api(check_alert.end).to_nanoseconds() if check_alert.end is not None else None,
+            priority=checklist._conjure_priority_to_priority(check_alert.priority)
+            if check_alert.priority is not scout_checks_api.Priority.UNKNOWN
+            else None,
+        )
+
+
+@dataclass(frozen=True)
+class DataReviewBatchBuilder:
+    integration_rids: list[str]
+    _requests: list[scout_datareview_api.CreateDataReviewRequest]
+    _clients: DataReview._Clients = field(repr=False)
+
+    def add_integration(self, integration_rid: str) -> DataReviewBatchBuilder:
+        self.integration_rids.append(integration_rid)
+        return self
+
+    def add_request(self, run_rid: str, checklist_rid: str, commit: str) -> DataReviewBatchBuilder:
+        self._requests.append(scout_datareview_api.CreateDataReviewRequest(checklist_rid, run_rid, commit))
+        return self
+
+    def initiate(
+        self, wait_for_completion: bool = True, wait_timeout: timedelta = timedelta(minutes=1)
+    ) -> Sequence[DataReview]:
+        """Initiates a batch data review process.
+
+        Args:
+            wait_for_completion (bool): If True, waits for the data review process to complete before returning.
+                                        Default is True.
+            wait_timeout (timedelta): The maximum time to wait for the data review process to complete.
+                                      Default is 1 minute.
+
+        Raises:
+            TimeoutError: If the data review process does not complete before the wait_timeout.
+        """
+        request = scout_datareview_api.BatchInitiateDataReviewRequest(
+            notification_configurations=[
+                scout_integrations_api.NotificationConfiguration(c) for c in self.integration_rids
+            ],
+            requests=self._requests,
+        )
+        response = self._clients.datareview.batch_initiate(self._clients.auth_header, request)
+
+        if not wait_for_completion:
+            return [
+                DataReview._from_conjure(self._clients, self._clients.datareview.get(self._clients.auth_header, rid))
+                for rid in response.rids
+            ]
+
+        started = datetime.now()
+        completed_review_rids = []
+        completed_reviews = []
+        while datetime.now() - started <= timedelta(seconds=wait_timeout.total_seconds()):
```

Comment:
```suggestion
        while datetime.now() - started <= wait_timeout:
```

no need to put it into a new timedelta

Reply:
:facepalm: